### PR TITLE
HV-1709 Polish Identification numbers are not considering length of the value

### PR DESCRIPTION
**Pull request by marko-bekhta**
- https://hibernate.atlassian.net/browse/HV-1709

Check that Polish Identification numbers are of correct length:
- Check the length of validated identification number
- Add tests for short/long values
- Add tests for nondigits in the values
- Disallow nondigits in PESEL

### CODE REVIEW

**Comment by michaldo** [comment]
I see you decided to keep existing design and fix defect with minimal change. OK.

I see also you decided to uniform logic and accept only digits for all three PESEL/NIP/REGON. That is also OK.
If you plan extension I would like to say that in Poland PESEL and REGON has always digit-only format, but NIP often exists as digit groups separated by space or dash, for example 836 15 73 483

Someone could be interested in less strict validation. But that is of course different story

**Comment by yrodiere** [comment]
Jenkins, retest this please.

File: engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/NIPValidatorTest.java [hunk 1]
```diff
@@ -24,10 +24,26 @@
...
 	@Test
-	public void testCorrectNipNumber() {
+	public void testAdditionalCharactersAreAllowed() {
 		assertNoViolations( validator.validate( new Person( "123-456-78-19" ) ) );
 		assertNoViolations( validator.validate( new Person( "123-45-67-819" ) ) );
 		assertNoViolations( validator.validate( new Person( "123-456-32-18" ) ) );
+	}
+
+	@Test
+	public void testIncorrectLenght() {
```
Comment (gsmet) [inline engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/NIPValidatorTest.java]: s/Lenght/Length/

File: engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/PESELValidatorTest.java [hunk 2]
```diff
@@ -23,6 +23,30 @@
...
+						violationOf( PESEL.class ).withProperty( "pesel" )
+				);
+		assertThat( validator.validate( new Person( "44-0-5-1-4-01359" ) ) )
+				.containsOnlyViolations(
+						violationOf( PESEL.class ).withProperty( "pesel" )
+				);
+	}
+
+	@Test
+	public void testIncorrectLenght() {
```
Comment (gsmet) [inline engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/PESELValidatorTest.java]: s/Lenght/Length/

File: engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/REGONValidatorTest.java [hunk 3]
```diff
@@ -23,6 +23,30 @@
...
+						violationOf( REGON.class ).withProperty( "regon" )
+				);
+		assertThat( validator.validate( new Company( "6-9-1-6-5-7-1-8-2" ) ) )
+				.containsOnlyViolations(
+						violationOf( REGON.class ).withProperty( "regon" )
+				);
+	}
+
+	@Test
+	public void testIncorrectLenght() {
```
Comment (gsmet) [inline engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/pl/REGONValidatorTest.java]: s/Lenght/Length/

**Comment by gsmet** [comment]
BTW, this should be backported to the 6.0 branch too.

**Comment by marko-bekhta** [comment]
merged and backported to 6.0. Thanks for reviews!

**Pull request closed without merging**